compiler/rustc_privacy/src/lib.rs

```diff
@@ -2303,26 +2200,8 @@ fn check_private_in_public(tcx: TyCtxt<'_>, (): ()) {
     };
     tcx.hir().walk_toplevel_module(&mut visitor);
 
-    let mut old_error_set_ancestry = HirIdSet::default();
```

Comment:
It should be possible to remove the whole `ObsoleteVisiblePrivateTypesVisitor` pass now.